lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala

```diff
@@ -0,0 +1,61 @@
+package org.ergoplatform.appkit.impl
+
+import java.util
+
+import org.ergoplatform._
+import org.ergoplatform.appkit.Parameters.{MinChangeValue, MinFee}
+import org.ergoplatform.appkit._
+
+import scala.collection.IndexedSeq
+import scala.language.postfixOps
+
+object TransactionOperations {
+
+  def buildUnsignedErgoTx(inputs: IndexedSeq[ErgoBox],
+                          dataInputs: IndexedSeq[DataInput],
+                          outputCandidates: Seq[ErgoBoxCandidate],
+                          feeAmount: Long,
+                          changeAddress: ErgoAddress,
+                          currentHeight: Int): UnsignedErgoLikeTransaction = {
+
+    require(feeAmount > 0, "Fee amount should be defined")
+    val inputTotal = inputs.map(_.value).sum
+    val outputSum = outputCandidates.map(_.value).sum
+    val outputTotal = outputSum + feeAmount
+    val changeAmt = inputTotal - outputTotal
+    val noChange = changeAmt < MinChangeValue
+    // if computed changeAmt is too small give it to miner as tips
+    val actualFee = if (noChange) feeAmount + changeAmt else feeAmount
+    require(
+      actualFee >= MinFee,
+      s"Fee must be greater then minimum amount ($MinFee NanoErg)"
+    )
+    val feeOut = new ErgoBoxCandidate(
+      actualFee,
+      ErgoScriptPredef.feeProposition(Parameters.MinerRewardDelay),
+      currentHeight
+    )
+    val changeOut =
+      new ErgoBoxCandidate(changeAmt, changeAddress.script, currentHeight)
+
+    val addedChangeOut = if (!noChange) Seq(changeOut) else Seq()
+
+    val finalOutputCandidates = outputCandidates ++ Seq(feeOut) ++ addedChangeOut
+
+    val mintedTokensNum = finalOutputCandidates
+      .flatMap(_.additionalTokens.toArray)
+      .count(t => util.Arrays.equals(t._1, inputs.head.id))
+
+    require(
+      mintedTokensNum <= 1,
+      s"Only one token can be minted, but found $mintedTokensNum"
+    )
+
+    val tx = new UnsignedErgoLikeTransaction(
+      inputs.map(b => new UnsignedInput(b.id)),
+      dataInputs,
+      finalOutputCandidates.toIndexedSeq
+    )
+    tx
+  }
+}
```

Comment:
@greenhat this code looks like belonging to sigmastate library, i.e. it mostly uses Sigma classes.
But I wonder how many of such methods we will need on Sigma side? There needs to be some systematic approach for creating them. And also how to make this methods stable, i.e. they should cover all possible transactions, withing our underlying transaction model.

Reply:
@aslesarenko I'd rather refrain from adding stuff to sigma, that sigma itself does not need. It's already huge.
We can put them into separate lib, or add to the wallet(we already have various box operations there).

Reply:
Ah, yes wallet is better. It looks also related to the issue https://github.com/ergoplatform/ergo/issues/1029